app/controllers/local_transaction_controller.rb

```diff
@@ -0,0 +1,129 @@
+class LocalTransactionController < ApplicationController
+  include ActionView::Helpers::TextHelper
+
+  before_filter :validate_slug_param, only: :search
+  before_filter -> { setup_content_item_and_navigation_helpers("/" + params[:slug]) }
+  before_filter -> { response.headers['X-Frame-Options'] = 'DENY' }
+  before_filter -> { set_expiry unless viewing_draft_content? }
+
+  INVALID_POSTCODE = 'invalidPostcodeFormat'.freeze
+  NO_AUTHORITY_URL = 'laMatchNoLinkNoAuthorityUrl'.freeze
+  NO_LINK = 'laMatchNoLink'.freeze
+  NO_MAPIT_MATCH = 'fullPostcodeNoMapitMatch'.freeze
+  NO_MATCHING_AUTHORITY = 'noLaMatch'.freeze
+
+  def search
+    @publication = publication
+    @edition = params[:edition]
+
+    if request.post?
+      @postcode = postcode
+      la_slug = local_authority_slug
+
+      return redirect_to local_transaction_results_path(local_authority_slug: la_slug) if la_slug
+    end
+
+    respond_to do |format|
+      format.html { render }
+      format.json { render json: @publication.to_json }
```

Comment:
We're redirecting to '/api/slug.json' for the other formats, I'm pretty sure we could do the same here so that we're consistent. Also I think this only returns an empty `places` hash so isn't really useful.